### PR TITLE
✨  add overall mean function

### PR DESCRIPTION
**Pull request by tpoisot**
This adds the overall mean lattice-wombling value from https://cdnsciencepub.com/doi/10.1139/x97-156 -- @TanyaS08 , this may be a good idea to substitute this in the land use example, so that we can use pretty-much all the information? I'm happy to take a first stab at a vignette.

### CODE REVIEW

**Comment by TanyaS08** [comment]
We should be able to do this with objects of type `TriangulationWomble` as well right?

**Comment by TanyaS08** [comment]
First I've created a new type/sub-type (MeanWomble) which is the output of `omlwv` (new name pending). I like this but we can opt for a no and rather keep to either lattice/triangualtion type
Second Instead of restricting `omlwv` to type LatticeWomble its now type Womble i.e. can simply pop either a vector of triangulation/lattice wobble items in. My only question is how 'safe' this is since you could theoretically input a mixed vector of types (I know we check that co-ords match but what if I stack a LatticeWomble & MeanWomble item that has the same co-ords??) So we _could_ go back to being more specific RE sub-types?

**Comment by TanyaS08** [comment]
Also at first pass the `boundaries` function seems to be a-okay with taking in and spitting out the MeanWomble item...

**Comment by TanyaS08** [comment]
Wait... I think I messed up on my 'clever' typing. Back to the drawing board....

**Comment by TanyaS08** [comment]
Okay back to rather having `omlwv` spit out either triangulation or lattice type (depending on input vector) so that they behave as you would expect for the different type.

File: src/lib/overallmean.jl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    for _idx in eachindex(w[1].m)
+        ch = filter(!isnan, [womble.m[_idx] for womble in w])
+        di = filter(!isnan, [deg2rad(womble.θ[_idx]) for womble in w])
+        if !isempty(ch)
+            m[_idx] = mean(ch)
+            α[_idx] = atan(mean(sin.(di)), mean(cos.(di)))
+        end
+    end
+    average_direction = rad2deg.(α) .+ 180.0
+    return LatticeWomble(m, average_direction, w[1].x, w[1].y)
```
Comment (tpoisot) [inline src/lib/overallmean.jl:25]: This needs to be `T`, not always a lattice womble

**Review by tpoisot — COMMENTED** [review]
the return type of `mean` is wrong

**Comment by tpoisot** [comment]
When this passes, you can merge!

**Pull request merged**